File: lms-FE/src/features/notifications/schemas.ts
import { z } from 'zod';

export const notificationSchema = z.object({
    id: z.number(),
    user_id: z.number(),
    type: z.string(),
    message: z.string(),
    is_read: z.boolean(),
    created_at: z.string(),
});

export type Notification = z.infer<typeof notificationSchema>;

File: lms-BE/app/features/notifications/schemas.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class NotificationCreate(BaseModel):
    user_id: int
    type: str
    message: str
    entity_id: Optional[int] = None

class NotificationRead(BaseModel):
    id: int
    user_id: int
    type: str
    message: str
    is_read: bool
    created_at: datetime

    class Config:
        from_attributes = True
